substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -737,18 +757,20 @@ impl<T: Config> Pallet<T> {
 
         let farm = pallet_tfgrid::Farms::<T>::get(node.farm_id).ok_or(Error::<T>::FarmNotExists)?;
 
-        if farm.dedicated_farm && !ActiveRentContractForNode::<T>::contains_key(node_id) {
-            return Err(Error::<T>::NodeNotAvailableToDeploy.into());
-        }
-
-        // If the user is trying to deploy on a node that has an active rent contract
-        // only allow the user who created the rent contract to actually deploy a node contract on it
+        let mut is_node_owner = false;
         if let Some(contract_id) = ActiveRentContractForNode::<T>::get(node_id) {
             let rent_contract =
                 Contracts::<T>::get(contract_id).ok_or(Error::<T>::ContractNotExists)?;
-            if rent_contract.twin_id != twin_id {
-                return Err(Error::<T>::NodeHasRentContract.into());
-            }
+            is_node_owner = rent_contract.twin_id == twin_id;
```

Comment:
is_contract_owner or owns_rent_contract would be a better name for the variable no?